pkg/operator/ceph/cluster/osd/create.go

```diff
@@ -381,5 +398,54 @@ func createDaemonOnNode(c *Cluster, osd OSDInfo, nodeName string, config *provis
 	updateConditionFunc(c.clusterInfo.Context, c.context, c.clusterInfo.NamespacedName(), k8sutil.ObservedGenerationNotAvailable, cephv1.ConditionProgressing, v1.ConditionTrue, cephv1.ClusterProgressingReason, message)
 
 	_, err = k8sutil.CreateDeployment(c.clusterInfo.Context, c.context.Clientset, d)
-	return errors.Wrapf(err, "failed to create deployment for OSD %d on node %q", osd.ID, nodeName)
+	if err != nil {
+		return errors.Wrapf(err, "failed to create deployment for OSD %d on node %q", osd.ID, nodeName)
+	}
+
+	if c.spec.Network.MultiClusterService.Enabled {
+		updatedDeployment, err := updateDeploymentArgs(c, osd, d)
+		if err != nil {
+			return errors.Wrapf(err, "failed to update external IP in deployment for OSD %d on node %q", osd.ID, nodeName)
+		}
+
+		_, err = k8sutil.CreateOrUpdateDeployment(c.clusterInfo.Context, c.context.Clientset, updatedDeployment)
```

Comment:
If there is no change to the deployment spec, K8s will basically ignore the update and not restart the pod. But it would be nice to skip the update call if it's an existing OSD. We only need to update a newly created OSD, right? Or maybe we're already in the code path that only creates new OSDs, so it's not an issue.